diff --git a/python/triton_kernels/triton_kernels/testing.py b/python/triton_kernels/triton_kernels/testing.py
index af56b1031..b8355728c 100644
--- a/python/triton_kernels/triton_kernels/testing.py
+++ b/python/triton_kernels/triton_kernels/testing.py
@@ -234,8 +234,8 @@ def alloc_rand(shape, device, dtype, requires_grad=False):
         tmp = 2**-(torch.randint(4, 8, shape, device=device, dtype=torch.float16))
         return tmp.to(dtype).requires_grad_(requires_grad)
     ret = torch.randn(shape, device=device, dtype=dtype, requires_grad=requires_grad)
-    ret = normalize_blocks(ret)
-    return ret
+    ret = normalize_blocks(ret.cpu())
+    return ret.to(device)


 def make_slice_sizes(n_slices, total_size, device="cuda"):
diff --git a/python/triton_kernels/tests/test_matmul.py b/python/triton_kernels/tests/test_matmul.py
index 813638969..5492cb4a9 100644
--- a/python/triton_kernels/tests/test_matmul.py
+++ b/python/triton_kernels/tests/test_matmul.py
@@ -433,7 +433,7 @@ def _test_op(m, n, k, split_k, do_gather, do_scatter, inner_expt_opt, do_gamma,
         maxtol, rmstol = 4e-1, 4e-2
     elif b_dtype.is_mxfloat4:
         maxtol, rmstol = 3e-2, None
-    assert_close(ref_y, tri_y, maxtol=maxtol, rmstol=rmstol)
+    assert_close(ref_y.cpu(), tri_y.cpu(), maxtol=maxtol, rmstol=rmstol)
     if c_dtype.has_global_scale:
         assert torch.all((ref_y_scale - tri_y_scale).abs() < 1e-10), \
                f"ref_y_scale: {ref_y_scale}, tri_y_scale: {tri_y_scale.item()}"
